DOCS/hydra-how-to-create-oauth-client.md

```diff
@@ -0,0 +1,67 @@
+# How to onboard an application with Charmed Hydra
+
+> This guide assumes that your application supports OAuth 2.0 and OpenID Connect (OIDC) protocols.
+> If your application has not yet adopted these standards, you’ll first need to:
+> - Enable your application to make an authorization request to an OAuth Server.
+> - Implement a callback endpoint (`redirect_uri`) that captures the authorization code received from the OAuth 2.0 Server (Hydra) and exchanges it for tokens.
+
+[note]
+
+The recommended libraries include:
+- Go: [oauth2](https://pkg.go.dev/golang.org/x/oauth2) and [oidc](https://pkg.go.dev/github.com/coreos/go-oidc/v3/oidc)
+- Python: [authlib](https://github.com/authlib/authlib), [oauthlib](https://github.com/oauthlib/oauthlib), [pyjwt](https://github.com/jpadilla/pyjwt), [python-jose](https://github.com/mpdavis/python-jose)
+
+[/note]
+
+To let your application authenticate with the Identity Platform, you must register it as an OAuth client in Hydra - the OAuth 2.0 and OIDC Server.
+
+You can do this in one of two ways:
+- by integrating your charmed application with [`oauth`](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/oauth/v0) interface
+- or using the `create-oauth-client` Juju action provided by [Charmed Hydra](https://charmhub.io/hydra).
+
+> The preferred approach is to integrate your charmed application with Hydra through the `oauth` interface, which automatically manages client registration based on the provided configuration.
+
+The Juju action and `oauth` interface offer the following parameters:
+- `audience`: A list of the audiences your client is allowed to request.
+- `grant-types`: A list of allowed grant types. Default: `["authorization_code"]`
+- `redirect-uris`: A list of your application's callback URLs that Hydra is allowed to redirect to.
+- `response-types`: A list of the allowed response types for the client. Default: `["code"]`
+- `scope`: A list of scopes the client is allowed to request. Default: `["openid", "profile", "email", "phone"]`
+- `token-endpoint-auth-method`: The authentication method the client is allowed to use at the token endpoint. One of `client_secret_post`, `client_secret_basic`
```

Comment:
I think public clients are also allowed, should we include `None` in this list? We may have some public clients that we want to onboard, should we add a section about that?